### ISSUE TITLE
get_clipboard_string() is broken

### ISSUE THREAD

**Issue by installgentoo** [PistonDevelopers/glfw-rs#474]
If you call get_clipboard_string() while you clipboard is empty, on linux at least, you'll panic on
```
    if (!*selectionString)
    {
        _glfwInputError(GLFW_FORMAT_UNAVAILABLE,
                        "X11: Failed to convert selection to string");
    }
```
and one needs to provide error callback to avoid this. Meanwhile get_clipboard_string() returns Option<> so you expect it to just return None instead of panicking.

**Comment by aloucks** [comment]
I agree that it's a little surprising, but this is the expected outcome if you use `glfw::init(glfw::FAIL_ON_ERRORS)`. Perhaps you could submit a [GLFW](https://github.com/glfw/glfw) issue to treat an empty clipboard differently from conversion error. 

As you mentioned, providing a custom error callback could be used to ignore `GLFW_FORMAT_UNAVAILABLE`. It wouldn't hurt to add some documentation to this method to describe this edge case.
